streamer/src/nonblocking/quic.rs

```diff
@@ -700,6 +720,33 @@ async fn packet_batch_sender(
     }
 }
 
+fn track_streamer_fetch_packet_performance(
+    packet_batch: &PacketBatch,
+    packet_perf_measure: &mut HashMap<[u8; 64], Instant>,
+    stats: Arc<StreamStats>,
+) {
+    for packet in packet_batch.iter() {
+        if packet.meta().is_perf_track_packet() {
+            let signature = get_signature_from_packet(packet);
+            if let Ok(signature) = signature {
+                if let Some(start_time) = packet_perf_measure.remove(signature) {
+                    let duration = Instant::now().duration_since(start_time);
+                    debug!(
+                        "QUIC streamer fetch stage took {duration:?} for transaction {:?}",
+                        Signature::from(*signature)
+                    );
+                    stats
+                        .process_sampled_packets_us_hist
+                        .lock()
```

Comment:
This seems like a lot of frequent short locks, should we lock and do the whole batch?

Reply:
Will look.

Reply:
Done